homeassistant/components/webostv/media_player.py

```diff
@@ -331,13 +349,41 @@ async def async_turn_on(self):
     @cmd
     async def async_volume_up(self):
         """Volume up the media player."""
+        if self._should_sleep_between_consecutive_volume_steps():
+            await self._sleep_between_consecutive_volume_steps()
         await self._client.volume_up()
 
     @cmd
     async def async_volume_down(self):
         """Volume down media player."""
+        if self._should_sleep_between_consecutive_volume_steps():
+            await self._sleep_between_consecutive_volume_steps()
         await self._client.volume_down()
 
+    def _should_sleep_between_consecutive_volume_steps(self):
+        """Return wheteher the medial player should sleep between consecutive volume steps."""
+        return self._consecutive_volume_steps_delay is not None
+
+    async def _sleep_between_consecutive_volume_steps(self):
+        """Sleep until the configured delay dusration has elapsed since the last volume step."""
+        async with self._last_volume_step_lock:
+            now = datetime.now()
+            sleep_time = timedelta(seconds=0)
+            if self._last_volume_step_datetime is not None:
+                since_last_volume_step = now - self._last_volume_step_datetime
+                delay = timedelta(milliseconds=self._consecutive_volume_steps_delay)
+                sleep_time = delay - since_last_volume_step
+
+                self._last_volume_step_datetime = now
+                if since_last_volume_step < delay:
+                    self._last_volume_step_datetime += sleep_time
+
+            else:
+                self._last_volume_step_datetime = now
+
+        if sleep_time.total_seconds() > 0:
+            await asyncio.sleep(sleep_time.total_seconds())
+
```

Comment:
i would move the sleep inside the lock. Now you get into a rather complex situation if a new request hit while you are sleeping on the previous. I see you increment the timestamp by sleep_time. I don't really see any benefit from sleeping in asyncio.sleep() rather than sleeping on the lock while a previous sleep is in progress.

Reply:
Your'e absolutely right. I made it much more complex than it needed to be. I changed it so that the action is performed, then i just sleep on the lock